magento/adobe-stock-integration
# License and save

### Description (*)
This PR introduces image license and save functionality. This PR also contains a set of refactorings since the approach that has been implemented for saving the preview images is not extensible as it is. 

### Fixed Issues (if relevant)
1. #401 : Licensing an image

### Manual testing scenarios (*)
1. Go to Admin -> Content -> Pages -> Choose a page | create new
2. Open dialog for inserting an image to a content
3. Click on "Open Adobe Stock"
4. Choose an image
5. Click on "License and Save" button
6. Click "Ok" in the confirmation window

The image should be saved to the Magento storage in its original size.
